spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -105,8 +102,8 @@
   private List<CombinedScanTask> tasks = null; // lazy cache of tasks
   private Boolean readUsingBatch = null;
 
-  Reader(Table table, Broadcast<FileIO> io, Broadcast<EncryptionManager> encryptionManager,
-      boolean caseSensitive, DataSourceOptions options) {
+  Reader(SparkSession spark, Table table, boolean caseSensitive, DataSourceOptions options) {
+    this.sparkContext = new JavaSparkContext(spark.sparkContext());
```

Comment:
This is valid, but it raises my heart rate. I would prefer we use JavaSparkContext.fromSparkContext so when I see this I don't think we are making a brand new context :)

Reply:
Good point. I did not know about `fromSparkContext`. Let me all places in this PR.

Reply:
Updated all places, resolving.